### PR TITLE
Feature/skale 4682 add versioning

### PR DESCRIPTION
**Pull request by yavrsky**

### CODE REVIEW

File: proxy/predeployed/scripts/generate_package_version.py [hunk 1]
```diff
@@ -22,14 +23,21 @@ def calculate_version(release_version):
...
+def getVersion():
+    version_key = 'VERSION'
+    if os.environ.get(version_key):
+        return os.environ.get(version_key)
+
+    try:
+        tag = subprocess.run("git describe --tags", shell=True, check=True, capture_output=True)
+        return tag.stdout.decode('utf-8')[:-1]
+    except:
+        with open('../../VERSION') as f:
```
Comment (payvint) [inline proxy/predeployed/scripts/generate_package_version.py:35]: I think we need to make it with try also - if no VERSION there - exit with message

File: proxy/predeployed/test/generate_genesis.py [hunk 2]
```diff
@@ -5,32 +5,36 @@
 
 
 def main():
-    if len(sys.argv) < 3:
+    if len(sys.argv) < 4:
```
Comment (payvint) [inline proxy/predeployed/test/generate_genesis.py]: It is an external change, we need to notify Skale-node about this change.

File: proxy/predeployed/test/generate_genesis.py [hunk 3]
```diff
@@ -5,32 +5,36 @@
...
-                    'deposit_box_erc1155_address': config['erc1155_deposit_box'],
-                    'linker_address': config['linker'],
-                    'community_pool_address': config['community_pool']
-                }))
-            print(json.dumps(genesis, indent=4, sort_keys=True))
+            with open(version_filename) as version_file:
+                base_genesis = json.load(base_genesis_file)
+                config = json.load(config_file)
+                genesis = base_genesis
+                version = version_file.readline().rstrip()
```
Comment (payvint) [inline proxy/predeployed/test/generate_genesis.py]: Or just get version of the component here

File: proxy/predeployed/test/generate_genesis.py (hunk 2 above)
Comment (gannakulikova) [reply proxy/predeployed/test/generate_genesis.py]: is this necessary? @payvint @yavrsky @DimaStebaev

**Review by payvint — DISMISSED** [review]

File: proxy/predeployed/src/ima_predeployed/contracts/message_proxy_for_schain.py [hunk 4]
```diff
@@ -56,15 +57,16 @@ class MessageProxyForSchainGenerator(ContractGenerator):
     IDX_HEAD = next_slot(OUTGOING_MESSAGE_DATA_HASH)
     IDX_TAIL = next_slot(IDX_HEAD)
     REGISTRY_CONTRACTS_SLOT = next_slot(IDX_TAIL)
+    VERSION_SLOT = next_slot(REGISTRY_CONTRACTS_SLOT)
     
 
-    def __init__(self, deployer_address: str, schain_name: str):
+    def __init__(self, deployer_address: str, schain_name: str, version: str):
```
Comment (DimaStebaev) [inline proxy/predeployed/src/ima_predeployed/contracts/message_proxy_for_schain.py]: Version is a representation of internal state. It useless to pass it outside.

**Review by DimaStebaev — CHANGES_REQUESTED** [review]

**Review by DimaStebaev — DISMISSED** [review]

**Review by payvint — APPROVED** [review]

**Review by DimaStebaev — APPROVED** [review]

**Review by sergiy-skalelabs — APPROVED** [review]

**Pull request merged**